### PR TITLE
[CELEBORN-1737] Support build tez client package

### PR DESCRIPTION
**Pull request by GH-Gloway**
### What changes were proposed in this pull request?
Add Tez packaging script.

### Why are the changes needed?
To support build tez client.

### Does this PR introduce _any_ user-facing change?
Yes, enable Celeborn with tez support. 

### How was this patch tested?
Cluster test.

### CODE REVIEW

**Comment by turboFei** [comment]
do you plan to have 2 modules likes other clients?
```
client-tez/tez
client-tez/tez-shaded
```

And could you enable the `tez` in https://github.com/apache/celeborn/blob/main/dev/dependencies.sh?

**Comment by FMX** [comment]
> do you plan to have 2 modules likes other clients?
> 
> ```
> client-tez/tez
> client-tez/tez-shaded
> ```
> 
> And could you enable the `tez` dependency check? https://github.com/apache/celeborn/blob/main/dev/dependencies.sh https://github.com/apache/celeborn/blob/main/.github/workflows/deps.yml
> 
> Also, it is better to provide the checking result for the shaded jar.
> 
> ```
> jar -tf $SHADED_JAR | grep -v celeborn | grep '\.class$'
> ```

I've updated this PR. 
The dep check for SBT will come up with a new PR.

File: build/make-distribution.sh [hunk 1]
```diff
@@ -248,6 +248,24 @@ function build_mr_client {
...
+        | tail -n 1)
+  BUILD_COMMAND=("$MVN" clean package $MVN_DIST_OPT -pl :celeborn-client-tez-shaded_${SCALA_VERSION} -am $@)
+
+    # Actually build the jar
+    echo -e "\nBuilding with..."
+    echo -e "\$ ${BUILD_COMMAND[@]}\n"
+
+    "${BUILD_COMMAND[@]}"
+
+    ## flink spark client jars
```
Comment (turboFei) [inline build/make-distribution.sh:264]: nit: seems typo

**Review by turboFei — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -1804,10 +1804,11 @@
       <id>tez</id>
       <modules>
         <module>client-tez/tez</module>
+        <module>client-tez/tez-shaded</module>
         <module>tests/tez-it</module>
       </modules>
       <properties>
-        <tez.version>0.10.3</tez.version>
+        <tez.version>0.10.2</tez.version>
```
Comment (turboFei) [inline pom.xml:1811]: by design？

**Review by FMX — APPROVED** [review]
LGTM. I've tested this PR on a cluster. Merged into main(v0.6.0).

**Pull request closed without merging**